### PR TITLE
v2 changes of PromptStudo-Registry

### PR DESCRIPTION
**Pull request by muhammad-ali-e**
## NOTE

- This PR is intended to make a copy of the existing applications to facilitate the transition to **v2**. It does not aim to change or optimize the existing code.
- Review comments can be added as **TODO** items for future enhancements.

## What

- Introduced v2 changes for the **prompt_studio.prompt_studio_registry** application.

## Why

- To align with the updated architecture and ensure compatibility with v2 applications.
- To streamline the handling of organization contexts without relying on django_tenants.

## How

- Updated imports to reference v2 instead of v1 applications.
- Revised the `get_organization` method to utilize the user context.
- Renamed application in `apps.py`.
- Added new `models` with updated table names.
- Updated `PromptStudioRegistry` model with `DefaultOrganizationMixin` class. 
- Added `PromptStudioRegistryModelManager` with `DefaultOrganizationManagerMixin` to get the organization data in `PromptStudioRegistry` of `prompt_studio_registry_v2`.

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- The new methods are self-contained ensuring no impact on current functionality.

## Database Migrations

- 

## Env Config

- 

## Relevant Docs

-

## Related Issues or PRs

1. https://github.com/Zipstack/unstract/pull/422
2. https://github.com/Zipstack/unstract/pull/430
3. https://github.com/Zipstack/unstract/pull/431
4. https://github.com/Zipstack/unstract/pull/432
5. https://github.com/Zipstack/unstract/pull/447
6. https://github.com/Zipstack/unstract/pull/461
7. https://github.com/Zipstack/unstract/pull/464
8. https://github.com/Zipstack/unstract/pull/465
9. https://github.com/Zipstack/unstract/pull/466
10. https://github.com/Zipstack/unstract/pull/467
11. https://github.com/Zipstack/unstract/pull/470
12. https://github.com/Zipstack/unstract/pull/471
13. https://github.com/Zipstack/unstract/pull/477

## Dependencies Versions

-

## Notes on Testing

-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Review by hari-kuriakose — APPROVED** [review]
@muhammad-ali-e Model looks good.

We can take care of duplication when we delete v1 eventually. Also, migration files can be generated together later.

**Review by gaya3-zipstack — APPROVED** [review]
Changes look fine.

File: backend/prompt_studio/prompt_studio_registry_v2/fields.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+class ToolVariablesJSONField(models.JSONField):
+    def from_db_value(self, value, expression, connection):  # type: ignore
+        metadata = super().from_db_value(value, expression, connection)
+        return metadata
+
+
+class ToolMetadataJSONField(models.JSONField):
+    def from_db_value(self, value, expression, connection):  # type: ignore
+        metadata = super().from_db_value(value, expression, connection)
+        return metadata
```
Comment (vishnuszipstack) [inline backend/prompt_studio/prompt_studio_registry_v2/fields.py:29]: @muhammad-ali-e can do some refactoring here to avoid code repetition?

**Review by vishnuszipstack — APPROVED** [review]
Left a NIT comment. others looks good.

File: backend/prompt_studio/prompt_studio_registry_v2/fields.py (hunk 1 above)
Comment (muhammad-ali-e) [reply backend/prompt_studio/prompt_studio_registry_v2/fields.py]: Updated with a base class along a TODO comment

**Review by harini-venkataraman — APPROVED** [review]

**Review by harini-venkataraman — APPROVED** [review]

**Pull request merged**